### PR TITLE
feat: Add two distinct endpoints for notes and name

### PR DESCRIPTION
**Pull request by rainandbare**
# Pull Request

## JIRA Ticket

[CZID-9003]

## Description

- Divided the `save_metadata` endpoint into 2 distinct endpoints. One for updating "sample_notes" and one for updating "name". 
- Also added a util function to check if you are logged in. 
- Also figured out how to have required fields in the request/response schemas 👯‍♀️

## Notes

Strangely, I couldn't figure out how to return anything other than the res object on success but that certainly is not imperative so I thought I would leave it in order to unblock Erin. 

## Tests

Test UpdateSampleName with:
```
mutation UpdateSampleName {
  UpdateSampleName(
    sampleId: [insert sampleId],
    input: {
      value:"A_New_Name"
      authenticityToken:[insert own token]
    }
  ){
    status
    message
  }
}
```

Test UpdateSampleNotes with:
```
mutation UpdateSampleNotes{
  UpdateSampleNotes(
    sampleId: [insert sampleId],
    input: {
      value:"Blah blah blah note note note"
      authenticityToken:[insert own token]
    }
  ){
    status
    message
  }
}
```

[CZID-9003]: https://czi-tech.atlassian.net/browse/CZID-9003?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ehoops-cz** [comment]
Screenshot 2023-12-15 at 1 25 37 PM
I'm using this branch locally and it looks like the input type for both note and name is set to `mutationInput_UpdateSampleNotes_input_Input`.  (this might be a "me" problem)

File: utils/httpUtils.ts [hunk 1]
```diff
@@ -44,6 +44,12 @@ export const getFullResponse = async (url: string, args: any, context: any) => {
     }
 };
 
+const checkForLogin = (responseUrl: string | null) => {
+    if (responseUrl?.includes("/auth0/refresh_token?mode=login")) {
+        throw new Error("You must be logged in to perform this action.");
+    }
+}
+
 export const postWithCSRF = async (url: string, body: any, args: any, context: any) => {
```
Comment (jfoo1984) [inline utils/httpUtils.ts:53]: thanks for adding this!

**Comment by rainandbare** [comment]
> Screenshot 2023-12-15 at 1 25 37 PM I'm using this branch locally and it looks like the input type for both note and name is set to `mutationInput_UpdateSampleNotes_input_Input`. (this might be a "me" problem)

Thats right - they both just take the 

> Screenshot 2023-12-15 at 1 25 37 PM I'm using this branch locally and it looks like the input type for both note and name is set to `mutationInput_UpdateSampleNotes_input_Input`. (this might be a "me" problem)

I did this on purpose because the input for both is just a value as a string and the auth token. The field is hard coded. Does that work for you?

**Comment by ehoops-cz** [comment]
> > Screenshot 2023-12-15 at 1 25 37 PM I'm using this branch locally and it looks like the input type for both note and name is set to `mutationInput_UpdateSampleNotes_input_Input`. (this might be a "me" problem)
> 
> Thats right - they both just take the
> 
> > Screenshot 2023-12-15 at 1 25 37 PM I'm using this branch locally and it looks like the input type for both note and name is set to `mutationInput_UpdateSampleNotes_input_Input`. (this might be a "me" problem)
> 
> I did this on purpose because the input for both is just a value as a string and the auth token. The field is hard coded. Does that work for you?

It works.  It was a little confusing when implementing - I think it would make more sense with a generic name if we could think of one.  Right now it just looks like someone did a copy/pasta on the frontend.

**Pull request merged**